swap/swap.go

```diff
@@ -308,27 +338,25 @@ func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	return balances, nil
 }
 
-// Peers returns a list of every peer known through SWAP
-func (s *Swap) Peers() (peers []enode.ID, err error) {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
-
+// BalancePeers returns a list of every peer known to have a balance set through SWAP
+func (s *Swap) BalancePeers() (peers []enode.ID, err error) {
 	knownPeers := make(map[enode.ID]bool)
 
-	// get peer IDs from store
-	storePeers, err := s.stateStore.Keys()
+	// get balance keys from store
+	storeBalancePeers, err := s.stateStore.Keys(balancePrefix)
 	if err != nil {
 		return nil, err
 	}
 
-	// add store peers to result and mark as present
-	for _, storePeer := range storePeers {
-		peerID := enode.HexID(storePeer)
+	// add peers with balance to result and mark as present
+	for _, storeBalancePeer := range storeBalancePeers {
+		// take balance key and turn into node ID
+		peerID := keyToID(storeBalancePeer, balancePrefix)
 		knownPeers[peerID] = true
 		peers = append(peers, peerID)
 	}
 
-	// add in-memory peers to result if not present
+	// add in-memory balance peers to result if not present
```

Comment:
This iteration will only add peers from memory if they were not already found on disk.
As we save the balances right away after every update, this is reasonable, but it also means that iterating over the memory balances is useless, as all entries should be found on disk too.

Maybe it's more efficient to have `known_peers` from memory, and only add those from disk if they were not in memory, what do you think?

Reply:
Done! Thanks for the suggestion.